operators/src/adapters/raster_subquery/raster_subquery_reprojection.rs

```diff
@@ -116,15 +104,50 @@ where
     }
 }
 
+fn build_accu<T: Pixel>(
+    query_rect: RasterQueryRectangle,
+    pool: Arc<ThreadPool>,
+    tile_info: TileInformation,
+    no_data_and_fill_value: T,
+    valid_bounds_out: Option<SpatialPartition2D>,
+    out_srs: SpatialReference,
+    in_srs: SpatialReference,
+) -> impl Future<Output = Result<TileWithProjectionCoordinates<T>>> {
+    tokio::task::spawn_blocking(move || {
+        let output_raster = EmptyGrid::new(tile_info.tile_size_in_pixels, no_data_and_fill_value);
+
+        let pool = pool.clone();
+
+        // if there is a valid output spatial partition, we need to reproject the coordinates.
+        let projected_coords = if let Some(valid_out_area) = valid_bounds_out {
+            projected_coordinate_grid_parallel(&pool, tile_info, out_srs, in_srs, &valid_out_area)?
+        } else {
+            debug!("reproject tile outside valid bounds"); // TODO: error?
+            Grid2D::new_filled(tile_info.tile_size_in_pixels, None, None)
+        };
+
+        Ok(TileWithProjectionCoordinates {
+            accu_tile: RasterTile2D::new_with_tile_info(
+                query_rect.time_interval,
+                tile_info,
+                output_raster.into(),
+            ),
+            coords: projected_coords,
+            pool,
+        })
+    })
+    .map_err(From::from)
+    .and_then(|x| async { x }) // flatten Ok(Ok())
+}
+
 fn projected_coordinate_grid_parallel(
     pool: &ThreadPool,
     tile_info: TileInformation,
     out_srs: SpatialReference,
     in_srs: SpatialReference,
     valid_out_area: &SpatialPartition2D,
 ) -> Result<Grid2D<Option<Coordinate2D>>> {
-    const OVERPROVISIONING: usize = 1; // split the work into multiple chunks per thread.
-    const MAX_PARALELLISM: usize = 8; // max task / work split
+    const MIN_ROWS_IN_PAR_CHUNK: usize = 64; // this must never be smaller than 1
```

Comment:
If you use min rows instead of pixels, doesn't this mean that if the tile size increases, you say that one thread needs to process more pixels in order to be effective? And if you lower the tile size, you say it needs less to process pixels in order to be effective.

I guess the number of pixels is also highly platform-dependent. It depends on the CPU clock, on SIMD register sizes, etc.

Do we find a robust way for this?

How does rayon decide on how much to schedule? Maybe we can do something similar.

Reply:
I use rows since that iis a very easy way to determine the pixel location. if we do it otherwise we have to assert that in each setup we use a number of pixels that is a multiple off a row lenght. I do not think we want to take that route. At least i don't want to implement that logic.
Yes, this also means that with larger tiles we have more pixels per thread but keep in mind that both axis increase when you change the tile size. We should never go below 512pix tiles.  In fact i think we should go up to 1024.

We will never find a perfect solution for all CPUs, CPU  Speed, CPU Arch,, RAM, HDDs, Network, .... We can determine what works and use that. I don't want to have any caps but i  guess the overhead of rayon is a think we have to deal with.

As i understand it rayon simply spawns as man threads as there are CPUs. If you use par_iter it will create one task per item. It simply does not decide how much work to schedule.

Reply:
fb4f5b4